File: platforms/swift/examples/scene/light/kind.swift
import FragmentColor

let sun = try Light.directional([0.0, -1.0, 0.0], [1.0, 1.0, 1.0])
let bulb = try Light.point([0.0, 2.5, 0.0], [1.0, 1.0, 1.0])
let torch = try Light.spot([0.0, 1.8, 1.0], [0.0, -1.0, 0.0], [1.0, 1.0, 1.0])

File: platforms/kotlin/examples/scene/light/kind.kt
import org.fragmentcolor.*

val sun = Light.directional(listOf(0.0f, -1.0f, 0.0f), listOf(1.0f, 1.0f, 1.0f))
val bulb = Light.point(listOf(0.0f, 2.5f, 0.0f), listOf(1.0f, 1.0f, 1.0f))
val torch = Light.spot(listOf(0.0f, 1.8f, 1.0f), listOf(0.0f, -1.0f, 0.0f), listOf(1.0f, 1.0f, 1.0f))